src/library_nodefs.js

```diff
@@ -232,18 +243,22 @@ addToLibrary({
       }
     },
     stream_ops: {
+      getattr(stream) {
+        return NODEFS.getattr(() => fs.fstatSync(stream.nfd));
+      },
+      setattr(stream, attr) {
+        NODEFS.setattr(stream.nfd, stream.node, attr, fs.fchmodSync, fs.futimesSync, fs.ftruncateSync);
+      },
       open(stream) {
         var path = NODEFS.realPath(stream.node);
         NODEFS.tryFSOperation(() => {
-          if (FS.isFile(stream.node.mode)) {
-            stream.shared.refcount = 1;
-            stream.nfd = fs.openSync(path, NODEFS.flagsForNode(stream.flags));
-          }
+          stream.shared.refcount = 1;
```

Comment:
So we are removing all of the `FS.isFile(stream.node.mode)` checks here.. are you sure that is OK / safe?

Reply:
Yeah, the new code paths require every NODEFS stream to have a `nfd`. Otherwise if we fstat the directory it will fail. I think this also fixes dup for nodefs directories -- I'm pretty sure it has the same problem I've fixed a bunch of times where if you open a directory, dup it, and then close it, attempting to continue using the original file descriptor will fail.